### ISSUE TITLE
Upgrade to PostgreSQL on Ubuntu 18.04

### ISSUE THREAD

**Issue by mvl22** [cyipt/cyipt-deploy#1]
Commit 53fbf4ad9f9f45b5acc090e67c1a80ae64ffb480 moved the repo to Ubuntu 18.04 (from 16.04), which works fine.

The OS upgrade produces the following notice about PostgreSQL:

```
 │ Obsolete major version 9.5                                                                                                    │  
 │                                                                                                                               │  
 │ The PostgreSQL version 9.5 is obsolete, but the server or client packages are still installed. Please install the latest      │  
 │ packages (postgresql-10 and postgresql-client-10) and upgrade the existing  clusters with pg_upgradecluster (see manpage).    │  
 │                                                                                                                               │  
 │ Please be aware that the installation of postgresql-10 will automatically create a default cluster 10/main. If you want to    │  
 │ upgrade the 9.5/main cluster, you need to remove the already existing 10 cluster (pg_dropcluster --stop 10 main, see manpage  │  
 │ for details).                                                                                                                 │  
 │                                                                                                                               │  
 │ The old server and client packages are no longer supported. After the existing clusters are upgraded, the postgresql-9.5 and  │  
 │ postgresql-client-9.5 packages should be removed.                                                                             │  
 │                                                                                                                               │  
 │ Please see /usr/share/doc/postgresql-common/README.Debian.gz for details.                                                     |
```

Note that the site still works fine after the upgrade - although 9.5 is "obsolete", it is still a valid and working package.

Upgrade notes for PostgreSQL 10 are at:  
https://www.postgresql.org/docs/10/pgupgrade.html  
https://www.paulox.net/2018/05/19/upgrading-postgresql-from-9-6-to-10-0-on-ubuntu-18-04/

In summary, this involves:

* Stop and drop the new cluster: `sudo pg_dropcluster 10 main --stop`
* Upgrading the old one: `sudo pg_upgradecluster 9.5 main`. Note that this will create a copy of the data in Postgres 10, meaning a 9GB disk space requirement, in parallel with the original v. 9.5 data.
* Then delete the old cluster (9.5 data): `sudo pg_dropcluster 9.5 main`.
* Then uninstall the postgresql-9.5/postgresql-client-9.5 packages: `sudo apt purge
postgresql-9.5 postgresql-client-9.5 postgresql-contrib-9.5`.

PostGIS then needs to be upgraded; this seems to be necessary in order to get the binary GIS structures rewritten. See:  
https://postgis.net/workshops/postgis-intro/upgrades.html#with-data-in-schemas

**Comment by Robinlovelace** [comment]
Thanks for the notice Martin. Great work on finding out what needs to be done. Should be a bit of a performance boost also I think.

**Comment by mvl22** [comment]
A simple fresh OS install, with the install script as updated in be9a1c3923df66d42fc4b6960bf4c3c1ae0a100e, and then importing a 9.5 dump works fine.
